include/irr/video/IGPUVirtualTexture.h

```diff
@@ -113,12 +112,11 @@ class IGPUVirtualTexture final : public asset::IVirtualTexture<video::IGPUImageV
         //now copy from CPU counterpart resources that can be shared (i.e. just copy state) between CPU and GPU
         //and convert to GPU those which can't be "shared": page table and VT resident storages along with their images and views
 
-        m_pageTable = std::move(_driver->getGPUObjectsFromAssets(&_cpuvt->getPageTable(), &_cpuvt->getPageTable()+1)->front());
+        auto* cpuPgt = _cpuvt->getPageTable();
+        m_pageTable = std::move(_driver->getGPUObjectsFromAssets(&cpuPgt, &cpuPgt+1)->front());
 
         m_freePageTableLayerIDs = _cpuvt->getFreePageTableLayersStack();
-
-        memcpy(m_layerToViewIndexMapping->data(), _cpuvt->getLayerToViewIndexMapping().begin(), 
-            _cpuvt->getLayerToViewIndexMapping().size()*sizeof(decltype(m_layerToViewIndexMapping)::pointee::value_type));
+        _cpuvt->writeLayerToViewIndexLUTContents(m_layerToViewIndexMapping.data(), false);
```

Comment:
could have been a plain assignment

Reply:
that would require direct access to cpuvt->m_layerToViewIndexMapping

Reply:
nope
`m_layerToViewIndexMapping = _cpuvt->getLayerToViewIndexMapping(false);`